source/Account.cpp

```diff
@@ -371,9 +371,18 @@ const vector<Mortgage> &Account::Mortgages() const
 
 // Add a new mortgage for the given amount, with an interest rate determined by
 // your credit score.
-void Account::AddMortgage(int64_t principal)
+void Account::AddMortgage(int64_t principal, int term)
 {
-	mortgages.emplace_back(principal, creditScore);
+	mortgages.emplace_back(principal, creditScore, term);
+	credits += principal;
+}
```

Comment:
Make this function call the other one instead of duplicating it maybe? If you use the std::optional like suggested below you could even merge the two functions into one.

Reply:
They're such small functions that it hardly matters.